mpqp/core/instruction/gates/gate.py

```diff
@@ -354,3 +354,22 @@ def __repr__(self) -> str:
     nb_qubits = (  # pyright: ignore[reportIncompatibleMethodOverride, reportAssignmentType]
         1
     )
+
+    @classmethod
+    def range(cls, start_or_end: int, end: Optional[int] = None, step: int = 1):
+        """Apply the gate to a range of qubits.
+
+        Args:
+            start_or_end: If ``end`` is not defined, this value is treated as
+                the end value of the range, and the range starts from ``0``.
+                Otherwise, it is treated as the start value.
```

Comment:
u sure 'bout that buddy ?

Reply:
![image](https://github.com/user-attachments/assets/b9cca0ed-731c-4268-b43c-8dbd8118bda3)

Both are correct apparently, "starting point" looks more common, "start point" more UK. We can ask our english referent ^^

Reply:
I'd say that using start instead of starting here is probably a bit less correct, but sticks closer to the variable name, it is also matching the "end point" part (which could also be "ending point" I guess)